library/std/unstable_arithmetic.qs

```diff
@@ -511,4 +512,223 @@ namespace Microsoft.Quantum.Unstable.Arithmetic {
         }
     }
 
-}
\ No newline at end of file
+    //
+    // Compare BigInt and qubit register in a little-endian format
+    //
+    //  target ^= c < x  | InvertIfLessL
+    //  target ^= c <= x | InvertIfLessOrEqualL
+    //  target ^= c == x | InvertIfEqualL
+    //  target ^= c >= x | InvertIfGreaterOrEqualL
+    //  target ^= c > x  | InvertIfGreaterL
+    //
+
+    /// # Summary
+    /// Computes `target ^= (c < x)`, that is, inverts `target`
+    /// if a BigInt value `c` is less than the little-endian qubit register `x`
+    operation InvertIfLessL (c : BigInt, x : Qubit[], target : Qubit) : Unit is Adj {
+        CompareGreaterThanOrEqualConstant(false, X, c + 1L, x, target);
+    }
+
+    /// # Summary
+    /// Computes `target ^= (c <= x)`, that is, inverts `target`
+    /// if a BigInt value `c` is less or equal to the little-endian qubit register `x`
+    operation InvertIfLessOrEqualL (c : BigInt, x : Qubit[], target : Qubit) : Unit is Adj {
+        CompareGreaterThanOrEqualConstant(false, X, c, x, target);
+    }
+
+    /// # Summary
+    /// Computes `target ^= (c == x)`, that is, inverts `target`
+    /// if a BigInt value `c` is equal to the little-endian qubit register `x`
+    operation InvertIfEqualL (c : BigInt, xs : Qubit[], target : Qubit) : Unit is Adj + Ctl {
+        let bits = BigIntAsBoolArray(c, Length(xs));
+        within {
+            ApplyPauliFromBitString(PauliX, false, bits, xs);
+        } apply {
+            Controlled X(xs, target);
+        }
+    }
+
+    /// # Summary
+    /// Computes `target ^= (c >= x)`, that is, inverts `target`
+    /// if a BigInt value `c` is greater or equal to the little-endian qubit register `x`
+    operation InvertIfGreaterOrEqualL (c : BigInt, x : Qubit[], target : Qubit) : Unit is Adj {
+        CompareGreaterThanOrEqualConstant(false, X, c + 1L, x, target);
+        X(target);
```

Comment:
Isn't the inversion already happening when calling `CompareGreaterThanOrEqualConstant`?

Reply:
I updated the code and hopefully it's easier to understand now. I also set invertControl to true to avoid additional X on target.
The function is named InvertIfGreaterOrEqualL(c, x), or it should invert target if c >= x. The helper function it calls checks if x is greater or equal than constant (as the name suggests), so it checks x >= c. We use this helper function as follows: c>=x  ⇔  c+1>x  ⇔  x<c+1  ⇔ not(x>=c+1). So we add 1 to c and invert control.